sdk/test/trace/sampler_test.cc

```diff
@@ -0,0 +1,205 @@
+  // The default tracer has two elements in the sampling result attribute
+  auto tracer_mock = initTracer(spans_received, std::make_shared<MockSampler>());
+  {
+    tracer_mock->StartSpan("span 3", {{"attr1", 314159}, {"attr2", false}});
+  }
+  auto &span_data3 = spans_received->at(0);
+  // It has a total of four attributes: two from StartSpan and two from the sampler.
+  ASSERT_EQ(4, span_data3->GetAttributes().size());
+}
+
+TEST(ParentOrElseSampler, ShouldSample)
+{
+  ParentOrElseSampler sampler_off(std::make_shared<AlwaysOffSampler>());
+  ParentOrElseSampler sampler_on(std::make_shared<AlwaysOnSampler>());
+
+  // Set up parameters
+  opentelemetry::trace::TraceId trace_id;
+  opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+  using M = std::map<std::string, int>;
+  M m1 = {{}};
+  opentelemetry::trace::KeyValueIterableView<M> view{m1};
+  opentelemetry::sdk::trace::Sampler::SpanContext parent_context_sampled(true, true);
+  opentelemetry::sdk::trace::Sampler::SpanContext parent_context_nonsampled(true, false);
+
+  // Case 1: Parent doesn't exist. Return result of delegateSampler()
+  auto sampling_result = sampler_off.ShouldSample(nullptr, trace_id, "", span_kind, view);
+  auto sampling_result2 = sampler_on.ShouldSample(nullptr, trace_id, "", span_kind, view);
+
+  ASSERT_EQ(Decision::NOT_RECORD, sampling_result.decision);
+  ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result2.decision);
+
+  // Case 2: Parent exists and SampledFlag is true
+  auto sampling_result3 =
+      sampler_off.ShouldSample(&parent_context_sampled, trace_id, "", span_kind, view);
+  ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result3.decision);
+
+  // Case 3: Parent exists and SampledFlag is false
+  auto sampling_result4 =
+      sampler_on.ShouldSample(&parent_context_nonsampled, trace_id, "", span_kind, view);
+  ASSERT_EQ(Decision::NOT_RECORD, sampling_result4.decision);
+}
+
+TEST(ParentOrElseSampler, GetDescription)
+{
+  ParentOrElseSampler sampler(std::make_shared<AlwaysOffSampler>());
+  ASSERT_EQ("ParentOrElse{AlwaysOffSampler}", sampler.GetDescription());
+  ParentOrElseSampler sampler2(std::make_shared<AlwaysOnSampler>());
+  ASSERT_EQ("ParentOrElse{AlwaysOnSampler}", sampler2.GetDescription());
+}
+
+TEST(AlwaysOffSampler, ShouldSample)
+{
+	AlwaysOffSampler sampler;
+
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	auto sampling_result = sampler.ShouldSample(nullptr, trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::NOT_RECORD, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+}
+
+TEST(AlwaysOffSampler, GetDescription)
+{
+	AlwaysOffSampler sampler;
+
+	ASSERT_EQ("AlwaysOffSampler", sampler.GetDescription());
+}
+
+TEST(AlwaysOnSampler, ShouldSample)
```

Comment:
I'm fine with having those tests in separate files (`always_off_sampler_test.cc` and others).